torch/csrc/jit/autodiff.cpp

```diff
@@ -188,6 +195,8 @@ static value_set findAllRequiresGradNodes(
 
   for (Node * node : graph.nodes()) {
     if (std::none_of(node->inputs().begin(), node->inputs().end(), requires_grad)) continue;
+    if (node->kind() == aten::type_as && !requires_grad(node->inputs()[0])) continue;
```

Comment:
Why is this condition necessary? Wouldn't it be handled by the previous case?

Reply:
type_as has two inputs, the second of which (setting type) might require grad, but it still won't affect the output of type_as requiring grad.

Reply:
Can you please add a comment that says just that?